src/always-be-playing.js

```diff
@@ -135,6 +120,93 @@ export default class GapSkipper {
   }
 
   /**
+   * Handler for `waiting` events from the player
+   *
+   * @private
+   */
+  waiting_() {
+    let seekable = this.seekable();
+    let currentTime = this.tech_.currentTime();
+    let playlist = this.playlist();
+
+    if (this.tech_.seeking() || this.timer_ !== null) {
+      return;
+    }
+
+    if (this.checkFellOutOfLiveWindow_(playlist, seekable, currentTime)) {
+      return;
+    }
+
+    let buffered = this.tech_.buffered();
+    let nextRange = Ranges.findNextRange(buffered, currentTime);
+
+    if (this.checkVideoUnderflow_(nextRange, buffered, currentTime)) {
+      return;
+    }
+
+    this.checkGap_(nextRange, currentTime);
+  }
+
+  checkFellOutOfLiveWindow_(playlist, seekable, currentTime) {
+    if (!playlist ||
+        playlist.endList ||
+        !seekable.length ||
```

Comment:
Since seekable.start(0) will always be zero for VOD, I don't think you need to worry about whether the playlist is live or not which would eliminate the need to pass in `playlist` here.